x-pack/platform/test/onechat_api_integration/apis/builtin_tools.ts

```diff
@@ -19,6 +19,13 @@ export default function ({ getService }: FtrProviderContext) {
     describe(`DELETE /api/agent_builder/tools/{toolName}`, () => {
       it('should return 400 error when attempting to delete any read-only builtin system tool', async () => {
         for (const toolId of Object.values(platformCoreTools) as string[]) {
+          if (toolId === platformCoreTools.createVisualization) {
+            await supertest
+              .delete(`/api/agent_builder/tools/${toolId}`)
+              .set('kbn-xsrf', 'kibana')
+              .expect(404);
+            return;
+          }
```

Comment:
We probably don't need to test deletion on **every** platform tool here. Can you adapt the test to try deletion for a single one (e.g. the search tool) instead of looping on all platform tools? Will allow to remove the specific condition you added